packages/next/src/server/is-private-ip.ts

```diff
@@ -0,0 +1,17 @@
+import ipaddr from 'next/dist/compiled/ipaddr.js'
+
+export function isPrivateIp(ip: string): boolean {
+  if (ip.startsWith('[') && ip.endsWith(']')) {
+    ip = ip.slice(1, -1)
+  }
+  if (!ipaddr.isValid(ip)) {
+    return false
+  }
+  try {
+    const addr = ipaddr.parse(ip)
```

Comment:
if the IP is ipv6, should we convert addr to ipv4 so that something like `::ffff:8.8.8.8` would be considered public?